programs/clearing_house/src/lib.rs

```diff
@@ -1115,14 +1118,17 @@ pub mod clearing_house {
             return Err(ErrorCode::InsufficientCollateralForSettlingPNL.into());
         }
 
-        let market_position = &mut user.positions[position_index];
         let bank = &mut bank_map.get_quote_asset_bank_mut()?;
         let market = &mut market_map.get_ref_mut(&market_index)?;
 
         let oracle_price = oracle_map.get_price_data(&market.amm.oracle)?.price;
-        controller::position::update_cost_basis(market, market_position, oracle_price)?;
+        let max_pnl_to_settle = calculate_max_pnl_to_settle(&user.positions[position_index])?;
+        let (_, unrealized_pnl) = calculate_base_asset_value_and_pnl_with_oracle_price(
+            &user.positions[position_index],
+            oracle_price,
+        )?;
 
-        let user_unsettled_pnl = market_position.unsettled_pnl;
+        let user_unsettled_pnl = max_pnl_to_settle.min(unrealized_pnl);
```

Comment:
do we have an sdk function for this for ui?

Reply:
nah ill add